### PR TITLE
New README for 4.0

### PR DESCRIPTION
**Pull request by jdorfman**
A new README for 4.0. It Fixes the missing logo, adds Discord, and more.

## Test plan

* Spell check
* Check links

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,77 @@
...
+- [Sourcegraph Cloud](https://sourcegraph.com) lets you search over the open source ecosystem. [Search strings](https://sourcegraph.com/search?q=context:global+failed+to+ensure+HEAD+exists&patternType=literal), [search patterns](https://sourcegraph.com/search?q=context:global+lang:python+range%28len%28:%5B1%5D%29%29&patternType=structural), [search symbols](https://sourcegraph.com/search?q=context:global+type:symbol+lang:typescript+%28OA%7Coa%7COa%29uth+%5BHh%5Dandler+-file:%28%5E%7C/%29node_modules/+&patternType=regexp&case=yes) and [find references](https://sourcegraph.com/github.com/spf13/cobra@a684a6d7f5e37385d954dd3b5a14fc6912c6ab9d/-/blob/command.go?L221:19&subtree=true#tab=references) across your entire codebase and the open source world.
```
Comment (iskyOS) [inline README.md:66]: @jdorfman This is Sourcegraph.com. Sourcegraph Cloud is the new single-tenant enterprise product

File: README.md [hunk 2]
```diff
@@ -1,31 +1,77 @@
...
 - [Sourcegraph Enterprise](https://docs.sourcegraph.com/#getting-started) lets you run your own Sourcegraph instance in your own environment.
```
Comment (iskyOS) [inline README.md:67]: This is Sourcegraph self-hosted (https://handbook.sourcegraph.com/company-info-and-process/communication/content_guidelines/terminology_guidelines/)

File: README.md [hunk 3]
```diff
@@ -1,31 +1,77 @@
...
+  <b>Understand, fix, and automate across your codebase with Sourcegraph's code intelligence platform</b>
+</p>
+
+&nbsp;
+
+---
+
+## 4.0 Features
+
+### 🧠 Code intelligence: more useful, more times daily for every dev
```
Comment (iskyOS) [inline README.md:38]:
```suggestion
### 🧠 Code intelligence: uplevel your code search
```

File: README.md (hunk 1 above)
Comment (iskyOS) [reply README.md:66]:
```suggestion
- [Sourcegraph.com](https://sourcegraph.com) lets you search over the open source ecosystem. [Search strings](https://sourcegraph.com/search?q=context:global+failed+to+ensure+HEAD+exists&patternType=literal), [search patterns](https://sourcegraph.com/search?q=context:global+lang:python+range%28len%28:%5B1%5D%29%29&patternType=structural), [search symbols](https://sourcegraph.com/search?q=context:global+type:symbol+lang:typescript+%28OA%7Coa%7COa%29uth+%5BHh%5Dandler+-file:%28%5E%7C/%29node_modules/+&patternType=regexp&case=yes) and [find references](https://sourcegraph.com/github.com/spf13/cobra@a684a6d7f5e37385d954dd3b5a14fc6912c6ab9d/-/blob/command.go?L221:19&subtree=true#tab=references) across your entire codebase and the open source world.
```

File: README.md [hunk 4]
```diff
@@ -1,31 +1,77 @@
...
+* Quickly access answers within your codebase with a revamped reference panel 
+
+### 🏗️ High-leverage ways to improve your entire codebase
+* Make changes across all of your codebase at enterprise scale with server-side Batch Changes (beta)
+  * Run large-scale or resource-intensive batch changes without clogging your local machine
+  * Run large batch changes quickly by distributing them across an autoscaled pool of compute instances
+  * Get a better debugging experience with the streaming of logs directly into Sourcegraph.
+
+### ☁️ Dedicated Sourcegraph Cloud instances for enterprise
+* Sourcegraph Cloud now offers dedicated, single-tenant instances of Sourcegraph
```
Comment (iskyOS) [inline README.md:52]:
```suggestion
* Sourcegraph Cloud now offers dedicated, single-tenant instances of Sourcegraph for enterprise
```

File: README.md (hunk 2 above)
Comment (jdorfman) [reply README.md]: @iskyOS  can you please make the change again? I'm having trouble.

**Review by iskyOS — COMMENTED** [review]
Finally reviewed! All the 4.0 / feature items look good. The only thing I recommend changing is the deployments section, since Sourcegraph Self-hosted and Sourcegraph OSS are different products.

**Comment by jdorfman** [comment]
@iskyOS check now: https://github.com/sourcegraph/sourcegraph/tree/jd/readme-4.0#readme

I removed the section in question and rebuilt it by referencing the docs. If there are still issues, let's hop on a call and finish it. Thanks!

**Review by iskyOS — APPROVED** [review]
LGTM!

**Pull request merged**